harvester_e2e_tests/integration/test_vm_functions.py

```diff
@@ -101,14 +153,414 @@ def test_minimal_vm(api_client, image_id, unique_vm_name, wait_timeout):
 
 @pytest.mark.p0
 @pytest.mark.virtualmachines
-@pytest.mark.dependency(depends=["minimal_vm"])
+@pytest.mark.negative
+@pytest.mark.parametrize("resource", [dict(cpu=MAX), dict(mem=MAX), dict(disk=MAX),
+                                      dict(mem=MAX, cpu=MAX), dict(mem=MAX, cpu=MAX, disk=MAX)],
+                         ids=['cpu', 'mem', 'disk', 'mem-and-cpu', 'mem-cpu-and-disk'])
+def test_create_vm_no_available_resources(request, resource, api_client, image_id,
+                                          wait_timeout, unique_vm_name):
+    """Creates a VM with outlandish resources for varying elements (purposefully negative test)
+
+    Prerequisite:
+        Setting opensuse-image-url set to a valid URL for
+        an opensuse image.
+
+    Manual Test Doc(s):
+        - N/A
+
+    Args:
+        request (FixtureRequest): https://docs.pytest.org/en/7.1.x/_modules/_pytest/fixtures.html#FixtureRequest # noqa
+        resource (dict): dict of name(s) & value that can be deconstructed
+        api_client (HarvesterAPI): HarvesterAPI client
+        image_id (str): corresponding image_id from fixture
+        wait_timeout (int): seconds for wait timeout from fixture
+        unique_vm_name (str): string of unique vm name
+
+    Raises:
+        AssertionError: when vm can not be created, all vms should be allowed to be created
+
+    Steps:
+    1. build vm object specs for outlandish resource(s) under test
+    2. request to build the vm, assert that succeeds
+    3. check for conditions of guest not running and vm being unschedulable
+    4. delete vm and volumes
+
+    Expected Result:
+    - building vm with outlandish resource requests to be successful
+    - asserting that the status condition of the vm that is built to not be running
+    - asserting that the status condition of the vm that is built to be unschedulable
+    - assert deleting vm and volumes to be successful
+    """
+    unique_name_for_vm = f"{''.join(resource.keys())}-{unique_vm_name}"
+    overall_vm_obj = dict(cpu=1, mem=2, disk=10)
+    overall_vm_obj.update(resource)
+
+    vm = api_client.vms.Spec(overall_vm_obj['cpu'], overall_vm_obj['mem'])
+    vm.add_image("disk-0", image_id, size=overall_vm_obj.get('disk'))
+    code, data = api_client.vms.create(unique_name_for_vm, vm)
+    assert 201 == code, (code, data)
+
+    endtime = datetime.now() + timedelta(seconds=wait_timeout)
+    while endtime > datetime.now():
+        code, data = api_client.vms.get_status(unique_name_for_vm)
+        if 200 == code and len(data.get('status', {}).get('conditions', [])) > 1:
+            check_dict = {
```

Comment:
consider this with type hint:
```python
check_dict : dict = { ... }
```

usually, we will not add any type name when defining variables, as a ducking type language, we would more consider it's behavior like:
```python
checks = dict(...)  # with type hint: checks : dict = dict(...)
# `s` means it is iterable
checks['something'] = 'good'
# slice with string usually means it would fit Mapping protocol
```

Reply:
That's a great point :smile:  - I think perhaps I had named it that to keep track in scope while scaffolding the test.

It might be great to bring in type hinting with some tests at some point.
Something like:
```python
def headline(text: str, centered: bool = False) -> str:
    if not centered:
        return f"{text.title()}\n{'-' * len(text)}"
    else:
        return f" {text.title()} ".center(50, "o")
```
Having a quick way to know at a function level what the args types are can be pretty useful :smile: 

I've gone ahead and renamed it from an explicit data type to just a simple checks variable :smile: